common/workunit/workunit.cpp

```diff
@@ -2659,83 +2659,67 @@ cost_type aggregateCost(const IConstWorkUnit * wu, const char *scope, bool exclu
     }
 }
 
-//aggregate disk costs from top-level subgraphs (when scope specified) or workflows (scope not specified)
-cost_type aggregateDiskAccessCost(const IConstWorkUnit * wu, const char *scope)
+void StatisticsAggregator::loadExistingAggregates(IConstWorkUnit &workunit)
```

Comment:
80% of the StatisticsAggregator is related to the workunit. The code to generate the aggregates is already in jstats as part of the CStatisticCollection code. The suggested change will require moving just one function with 4 lines into jstats - everything else is wu related.

Reply:
You're right a lot of it is wuid related at the moment (mainly the loader), but as per other comments refreshAggregates should really be abstracted out of CStatisticCollection and into this class, and probably also a method that contains the logic of setting a stat. and ensuring scope (instead of IStatisticCollection::setStatatistic(const char *fullScope..,) (related to comment : https://github.com/hpcc-systems/HPCC-Platform/pull/18048#discussion_r1399645364).

The workunit specific parts could be implemented (in workunit.cpp) as standalone methods that utilize a StatisticsAggregator instance.

We can revisit later though, I would not refactor now.

Reply:
>void StatisticsAggregator::loadExistingAggregates(IConstWorkUnit &workunit)

picky: IConstWorkUnit could be a const IConstWorkUnit